diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter3/files/NewlinesRuleWarnTest.kt

```diff
@@ -0,0 +1,179 @@
+        )
+    }
+
+    @Test
+    fun `line breaking at operators`() {
+        lintMethod(NewlinesRule(),
+                """
+                    |fun foo() {
+                    |    val and = condition1
+                    |        && condition2
+                    |    // this isn't an expression
+                    |    val plus = x
+                    |        + y
+                    |       
+                    |    obj.
+                    |        foo()
+                    |    obj?.
+                    |        foo()
+                    |    obj ?:
+                    |        OBJ
+                    |    obj::
+                    |        foo
+                    |}
+                """.trimMargin(),
+                LintError(3, 9, ruleId, "$shouldBreakAfter &&", true),
+                LintError(8, 8, ruleId, "$shouldBreakBefore .", true),
+                LintError(10, 8, ruleId, "$shouldBreakBefore ?.", true),
+                LintError(12, 9, ruleId, "$shouldBreakBefore ?:", true),
+                LintError(14, 8, ruleId, "$shouldBreakBefore ::", true)
+        )
+    }
+
+    @Test
+    fun `chained calls should follow functional style - positive example`() {
+        lintMethod(NewlinesRule(),
+                """
+                    |fun foo(list: List<Bar>?) {
+                    |    list!!
+                    |        .filterNotNull()
+                    |        .map { it.baz() }
+                    |        .firstOrNull {
+                    |            it.condition()
+                    |        }
+                    |        ?.qux()
+                    |}
+                """.trimMargin()
+        )
+    }
+
+    @Test
+    fun `chained calls should follow functional style - should not trigger on single dot calls`() {
+        lintMethod(NewlinesRule(),
+                """
+                    |fun foo(bar: Bar?) {
+                    |    bar.baz()
+                    |    bar?.baz()
+                    |    bar!!.baz()
+                    |}
+                """.trimMargin()
+        )
+    }
+
+    @Test
+    fun `chained calls should follow functional style`() {
+        lintMethod(NewlinesRule(),
+                """
+                    |fun foo(list: List<Bar>?) {
+                    |    list!!.filterNotNull()
+                    |        .map { it.baz() }.firstOrNull {
+                    |            it.condition()
+                    |        }?.qux()
+                    |}
+                """.trimMargin(),
+                LintError(2, 11, ruleId, "$functionalStyleWarn .", true),
+                LintError(3, 26, ruleId, "$functionalStyleWarn .", true),
+                LintError(5, 10, ruleId, "$functionalStyleWarn ?.", true)
+        )
+    }
+
+    @Test
+    fun `chained calls should follow functional style - exception for ternary if-else`() {
+        lintMethod(NewlinesRule(),
+                """
+                    |fun foo(list: List<Bar>?) {
```

Comment:
lets add regression test for the future with the function and argument list?

Reply:
Yes, will add tests and mark them `@Disabled`.